packages/backend/src/tokens/request.ts

```diff
@@ -381,10 +394,36 @@ ${error.getFullMessage()}`,
     return signedOut(authenticateContext, AuthErrorReason.UnexpectedError);
   }
 
+
+  let requestState = null;
+
   if (authenticateContext.sessionTokenInHeader) {
-    return authenticateRequestWithTokenInHeader();
+    requestState = await authenticateRequestWithTokenInHeader();
   }
-  return authenticateRequestWithTokenInCookie();
+  else {
+    requestState = await authenticateRequestWithTokenInCookie();
+  }
+
+  if (options.organizationSync) {
+    const toActivate = getActivationEntity(authenticateContext.clerkUrl, options.organizationSync);
+    if (toActivate) {
+      const auth = requestState.toAuth()
+      if (auth) {
+        let mustActivate = false;
+        if (toActivate.organizationSlug && auth.orgSlug && toActivate.organizationSlug !== auth.orgSlug) {
+          mustActivate = true;
+        }
+        if (toActivate.organizationId && auth.orgId && toActivate.organizationId !== auth.orgId) {
+          mustActivate = true;
+        }
+        if (mustActivate) {
+          return handleMaybeHandshakeStatus(authenticateContext, AuthErrorReason.ActiveOrganizationMismatch, '');
+        }
+      }
+
+    }
+  }
+  return requestState
```

Comment:
I'm not love with doing all this work after the `authenticateRequestWithTokenIn...` functions run. It also seems fine to do this right before we call `signedIn`, which would allow us to avoid handshaking in the case that we just resolved a handshake, which might be important (i'll think that through).